### PR TITLE
add single event schema

### PR DESCRIPTION
**Pull request by kalkbrennerei**
**Proposed changes**:
- add single event schema so that it can be used to validate events in rasa x

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: changelog/10643.misc.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Add a schema for single events.
```
Comment (ancalita) [inline changelog/10643.misc.md:1]: Could you please add the motivation for this refactoring please, either in the changelog or the PR description?

File: tests/shared/utils/schemas/test_events.py [hunk 2]
```diff
@@ -27,4 +89,16 @@ def test_remote_action_validate_all_event_subclasses(event_class: Type[Event]):
         "wrong_action",
         "warning_predicted",
     ]:
-        jsonschema.validate(response, RemoteAction.action_response_format_spec())
+        validate(response, RemoteAction.action_response_format_spec())
+
+
+def test_validate_single_event():
```
Comment (ancalita) [inline tests/shared/utils/schemas/test_events.py:95]: Doesn't this [existing test](https://github.com/RasaHQ/rasa/blob/d7456ee04bd55d49468ee82a70a123a73f9fe6f8/tests/core/test_actions.py#L507) already check for valid events?

**Review by ancalita — COMMENTED** [review]
Looks good, just need some clarification on the refactoring motivation, also one of the tests seems to be a duplicate of an existing one for the same check.

File: changelog/10643.misc.md (hunk 1 above)
Comment (kalkbrennerei) [reply changelog/10643.misc.md]: good point!!

File: changelog/10643.misc.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Add a schema for single events so that it can be used for validation.
```
Comment (ancalita) [inline changelog/10643.misc.md:1]: Just needs a little bit more detail, since it's not very clear to my in which context would you need to use `EVENT_SCHEMA`  rather than `EVENTS_SCHEMA`? 😅

Comment (kalkbrennerei) [reply changelog/10643.misc.md:1]: haha yeah it seems kind of random, but I want to use `EVENT_SCHEMA` to validate message events in rasa x. That's why I also wanted to test the validation separately from the rasaos endpoint

**Review by ancalita — APPROVED** [review]
👍🏼
